application_base.cpp

```diff
@@ -152,8 +152,8 @@ void application_base::wait_for_signal(std::shared_ptr<boost::asio::signal_set>
    });
 }
 
-void application_base::setup_signal_handling_on_ios(boost::asio::io_service& ios, bool startup) {
-   std::shared_ptr<boost::asio::signal_set> ss = std::make_shared<boost::asio::signal_set>(ios, SIGINT, SIGTERM);
+void application_base::setup_signal_handling_on_ios(boost::asio::io_context& io_ctx, bool startup) {
```

Comment:
Since you're touching everything to rename `io_service`->`io_context`, maybe might as well rename `setup_signal_handling_on_ios` to `setup_signal_handling_on_ioc` or such, since `ios` meant **io**_**s**ervice :man_shrugging: